sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -434,75 +490,47 @@
      * or has been cancelled. The completed operation returns a List of {@link RecognizedForm}.
      * @throws FormRecognizerException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
      * an {@link OperationStatus#FAILED}.
-     * @throws NullPointerException If {@code recognizeOptions} is {@code null}.
+     * @throws NullPointerException If {@code receipt} is null.
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PollerFlux<OperationResult, List<RecognizedForm>>
         beginRecognizeReceipts(Flux<ByteBuffer> receipt, long length, RecognizeOptions recognizeOptions) {
         try {
+            Objects.requireNonNull(receipt, "'receipt' is required and cannot be null.");
+
             recognizeOptions = getRecognizeOptionsProperties(recognizeOptions);
+            final boolean isIncludeFieldElements = recognizeOptions.isIncludeFieldElements();
             return new PollerFlux<>(
                 recognizeOptions.getPollInterval(),
-                receiptStreamActivationOperation(receipt, length,
-                    recognizeOptions.getContentType(), recognizeOptions.isIncludeFieldElements()),
-                extractReceiptPollOperation(),
+                streamActivationOperation(
+                    (contentType -> service.analyzeReceiptAsyncWithResponseAsync(
+                        contentType, receipt, length, isIncludeFieldElements).map(response -> new OperationResult(
+                            parseModelId(response.getDeserializedHeaders().getOperationLocation())))),
+                    receipt, recognizeOptions.getContentType()),
+                pollingOperation(service::getAnalyzeReceiptResultWithResponseAsync),
                 (activationResponse, context) -> monoError(logger,
                     new RuntimeException("Cancellation is not supported")),
-                fetchExtractReceiptResult(recognizeOptions.isIncludeFieldElements()));
+                fetchingOperation(service::getAnalyzeReceiptResultWithResponseAsync)
+                    .andThen(after -> after.map(modelSimpleResponse -> {
+                        throwIfAnalyzeStatusInvalid(modelSimpleResponse.getValue());
```

Comment:
Throwing an exception inside a `map` operator is odd. In this case, use `handle()` operator to either `sink.error()` or `sink.next()` depending on whether the response is an error or success.

Reply:
the throwing an exception will be removed. The issue related to the handling failed status is here: https://github.com/Azure/azure-sdk-for-java/issues/12200

Reply:
@mssfang Currently, treat it as the exception will be thrown. Can update this later when we actually onboard the LRO failure handling changes.